code/game/objects/items/storage/firstaid.dm

```diff
@@ -517,3 +517,111 @@
 /obj/item/storage/pill_bottle/paxpsych/PopulateContents()
 	for(var/i in 1 to 5)
 		new /obj/item/reagent_containers/pill/paxpsych(src)
+
+/obj/item/storage/organbox
+	name = "organ transport box"
+	desc = "An advanced box with an cooling mechanism that uses cryostylane or other cold reagents to keep the organs or bodyparts inside preserved."
+	icon_state = "organbox"
+	lefthand_file = 'icons/mob/inhands/equipment/medical_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/medical_righthand.dmi'
+	throw_speed = 3
+	throw_range = 7
+	custom_premium_price = 1100
+	/// var to prevent it freezing the same things over and over
+	var/cooling = FALSE
+
+/obj/item/storage/organbox/ComponentInitialize()
+	. = ..()
+	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
+	STR.max_w_class = WEIGHT_CLASS_BULKY /// you have to remove it from your bag before opening it but I think that's fine
+	STR.max_combined_w_class = 21
+	STR.set_holdable(list(
+		/obj/item/organ,
+		/obj/item/bodypart,
+		/obj/item/reagent_containers/food/snacks/icecream
+		))
+
+/obj/item/storage/organbox/Initialize()
+	. = ..()
+	create_reagents(100, TRANSPARENT)
+	RegisterSignal(src, COMSIG_ATOM_ENTERED, .proc/freeze)
+	RegisterSignal(src, COMSIG_TRY_STORAGE_TAKE, .proc/unfreeze)
+	START_PROCESSING(SSobj, src)
+
+/obj/item/storage/organbox/process()
+	///if there is enough coolant var
+	var/cool = FALSE
+	var/amount = reagents.get_reagent_amount(/datum/reagent/cryostylane)
+	if(amount >= 0.1)
+		reagents.remove_reagent(/datum/reagent/cryostylane, 0.1)
+		cool = TRUE
+	else
+		amount = reagents.get_reagent_amount(/datum/reagent/consumable/ice)
+		if(amount >= 0.3)

```

Comment:
It seems weird that it'd consume ice if there was nothing to keep fresh. Means that if you wanted to have the box on hand, you'd also need a beaker full of ice (which would be perfectly inert in a beaker), and then when you transfered the body parts, you'd then also need to add the ice.

Reply:
its for emergency, you active it when you need it